bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchTestEnvironmentTaskProvider.scala

```diff
@@ -195,12 +196,22 @@ class BspFetchTestEnvironmentTaskProvider extends BeforeRunTaskProvider[BspFetch
       case moduleBasedRunConfig: ModuleBasedConfiguration[_, _]
         if BspUtil.isBspModule(moduleBasedRunConfig.getConfigurationModule.getModule) =>
         val classes = moduleBasedRunConfig match {
-          case p: ScalaTestRunConfiguration =>
-            p.testConfigurationData match {
+          case scalaTestConfig: ScalaTestRunConfiguration =>
+            scalaTestConfig.testConfigurationData match {
               case data: AllInPackageTestData => data.classBuf.asScala
               case data: ClassTestData => List(data.testClassPath)
               case _ => List()
             }
+          case junitConfig: JUnitConfiguration => {
+            junitConfig.getTestType match {
+              case JUnitConfiguration.TEST_METHOD => {
+                List(junitConfig.getPersistentData.getMainClassName)
+              }
+              case JUnitConfiguration.TEST_CLASS => {
+                List(junitConfig.getPersistentData.getMainClassName)
```

Comment:
the same code as for `TEST_METHOD`. intentional?

Reply:
We need to extract class name here, so yes. Btw I just noticed the default case was not handled here.

Reply:
Ok. I'd put it in the same condition then

Reply:
right!